### ISSUE TITLE
Integrating nocrypt?

### ISSUE THREAD

**Issue by pepa65** [fsmithred/refractasnapshot-base#4]
Using the Devuan Live iso, I never had to use nocrypt.sh (not sure why). But the Live iso doesn't have proper UTF-8 support, so I started working from a 'normal' (encrypted) Devuan install. I installed the -base deb-packages from Sourceforge, and discovered I now need to use nocrypt.sh because of my encrypted installation.

It would be way easier and entirely feasible to integrate nocrypt.sh into the main refractainstaller script. I will be doing this for myself anyways, would you be willing to accept a patch or pull request (or would you like to do this yourself)?

**Comment by fsmithred** [comment]
I usually integrate code manually when someone submits a patch. That's
because the newest version is always on my computer and gets put on git
later. Right now I'm away from home, but I'll look at it next week.

-fsr

**Comment by pepa65** [comment]
I basically added this function to `refractasnapshot` (after `set_boot_options` - both the function definition and the function call):

```bash
nocrypt () {
# Assuming that when /dev/mapper/devuan--vg-root is present, it is encrypted
# (it could be named differently, or it could just be unencrypted lvm2)
if [[ -e /dev/mapper/devuan--vg-root ]] ; then
	initrd_dir=$(mktemp -d)
	cd "$initrd_dir"
	fakeroot zcat /initrd.img | cpio -i
	
	if [[ -f conf/conf.d/cryptroot ]] ; then
		echo "Removing cryptroot from initrd"
		rm -f conf/conf.d/cryptroot
	fi
	if [[ -f conf/conf.d/resume ]] ; then
		echo "Removing resume from initrd"
		rm -f conf/conf.d/resume
	fi
	
	fakeroot find . -print0 | cpio -0 -H newc -o | gzip -c >initrd.img
	mv initrd.img "$work_dir"/iso/live/
	
	cd -
	rm -r -- "$initrd_dir"
fi
}
```

I was going to test extensively about root being on an encrypted partition, but within the context of the automated Devuan install it seemed enough. Since the script should in principle work on any install, the initial test in the function needs a lot more work. I was going to `grep ' / ' /proc/mounts` (or `grep ' on / type ' < <(mount)`) and then check against the output of `lsblk`. Would the snapshot also work without lvm2??

You are probably more aware of the constrainst and assumptions here...

**Comment by fsmithred** [comment]
Yeah, that's pretty much why I left it as a separate script. I'll play with
it when I get home next week.

I know the nocrypt script works with encrypted partitions without lvm.
That's what my setup is. And the mapper names vary. I've got root_fs,
home_fs, and sdXn_crypt. Another approach is to use blkid to find the
partitions with 'crypt' in the output, but I'm not certain if that works
all the time.

When it comes to partitions, my default position is to let the user
determine what's what. If the drive order changes or they have an unusual
setup, they are the only ones who can know that.

**Comment by pepa65** [comment]
In that case an `encrypted_root` yes/no setting in `/etc/refractasnapshot.conf` would be the most userfriendly and in line with current practice. I've really appreciated not having to run the `nocrypt.sh` manually in my setup though..!
